Python/test/ratehelpers.py

```diff
@@ -147,41 +148,21 @@ def build_eur_curve(self, quotes_date):
         self.oisSwapCurve.enableExtrapolation()
         self.discounting_yts_handle.linkTo(self.oisSwapCurve)
 
-    def test_ois_ratehelper_setTermStructure(self):
-        """Test if OISRateHelper.impliedQuote provides original quote from curve"""
-        for key, rate_helper in zip(self.ois.keys(), self.oisHelpers):
-            expected = self.ois[key].value()
-            # This test case is just to prove, that only if we provide
-            # explicitely the curve to the rate helper, the quote can be
-            # calculated
-            rate_helper.setTermStructure(self.oisSwapCurve)
-            # based on bootstrapped_curve
-            calculated = rate_helper.impliedQuote()
-            # I need to run calculation again - it looks as if the update of the
-            # curve was slow - if I don't do that, in the first weekly quote I
-            # get strange result of -0.006840684068403689
-            calculated = rate_helper.impliedQuote()
-            diff = (expected - calculated) * 1E4
-            print("Maturity: {}: diff: {} bps".format(rate_helper.maturityDate(),
-                                                      diff))
-            self.assertAlmostEqual(expected, calculated,
-                                   delta=1e-5,
-                                   msg="Calculated implied quote differes too "
-                                       "much from original market value")
-
     def test_ois_ratehelper_impliedquote(self):
         """Test if OISRateHelper.impliedQuote provides original quote from curve"""
+        # initiate curves - required due to lazy evaluation
+        df = self.discounting_yts_handle.discount(0.0)
+
```

Comment:
@lballabio thank you very much for the suggestion, that solves the problem.
The question is, why impliedQuote method, which actually calls for discount factors is not able to trigger lazy evaluation?